src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -359,7 +359,12 @@ private bool IsCountableAndIndexable(SyntaxNode node, TypeSymbol inputType, out
             }
             else
             {
-                hasErrors |= !TryBindLengthOrCount(node, receiverPlaceholder, out lengthAccess, diagnostics);
+                var foundLengthOrCount = TryBindLengthOrCount(node, receiverPlaceholder, out lengthAccess, diagnostics);
+                if (!foundLengthOrCount && !hasErrors)
```

Comment:
> && !hasErrors [](http://example.com/codeflow?start=39&length=14)

Why not always report the error?

Reply:
It looks like the only situation when ```hasErrors``` can be true here is when ```inputType.IsDynamic()```. Why even trying to locate Length/Count in that case, we know it is going to fail. It might make sense to refactor the code according to this, i.e. base the logic on the type, rather than on ```hasErrors```.